janusgraph-core/src/main/java/org/janusgraph/core/JanusGraph.java

```diff
@@ -77,6 +77,14 @@
         method = "shouldResetAfterRollback",
         reason = "JanusGraph assumes lifecycle of transactionListeners in AbstractThreadLocalTransaction ends when the " +
             "transaction ends (commit/rollback/close). TinkerPop, however, asserts transactionListeners are active across transactions.")
+@Graph.OptOut(
+        test = "org.apache.tinkerpop.gremlin.process.traversal.step.map.MergeVertexTest",
+        method = "g_mergeVXlabel_person_name_markoX_optionXonMatch_age_19X_option",
+        reason = "JanusGraph doesn't support MergeVertex yet.")
+@Graph.OptOut(
+        test = "org.apache.tinkerpop.gremlin.process.traversal.step.map.MergeVertexTest",
+        method = "g_withSideEffectXc_label_person_name_markoX_withSideEffectXm_age_19X_mergeVXselectXcXX_optionXonMatch_selectXmXX_option",
+        reason = "JanusGraph doesn't support MergeVertex yet.")
```

Comment:
We should at least open an issue for this feature to be implemented in JanusGraph

Reply:
Good idea

Reply:
https://github.com/JanusGraph/janusgraph/issues/3251